homeassistant/components/google/config_flow.py

```diff
@@ -0,0 +1,135 @@
+        self.external_data = info
+        return await super().async_step_creation(info)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle external yaml configuration."""
+        if not self._reauth and self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        await self.async_set_unique_id(DOMAIN)
+        return await super().async_step_user(user_input)
+
+    async def async_step_auth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Create an entry for auth."""
+        # The default behavior from the parent class is to redirect the
+        # user with an external step. When using the device flow, we instead
+        # prompt the user to visit a URL and enter a code. The device flow
+        # background task will poll the exchange endpoint to get valid
+        # creds or until a timeout is complete.
+        if user_input is not None:
+            return self.async_show_progress_done(next_step_id="creation")
+
+        if not self._device_flow:
+            _LOGGER.debug("Creating DeviceAuth flow")
+            try:
+                device_flow = await async_create_device_flow(self.hass)
+            except OAuthError as err:
+                _LOGGER.error("Error initializing device flow: %s", str(err))
+                return self.async_abort(reason="oauth_error")
+            self._device_flow = device_flow
+
+            async def _exchange_finished(creds: Credentials | None) -> None:
+                self.external_data = {"creds": creds}
+                self.hass.async_create_task(
+                    self.hass.config_entries.flow.async_configure(
+                        flow_id=self.flow_id, user_input={}
+                    )
+                )
+
+            await device_flow.start_exchange_task(_exchange_finished)
+
+        return self.async_show_progress(
+            step_id="auth",
+            description_placeholders={
+                "url": self._device_flow.verification_url,
+                "user_code": self._device_flow.user_code,
+            },
+            progress_action="exchange",
+        )
+
+    async def async_step_creation(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle external yaml configuration."""
+        if self.external_data.get("creds") is None:
+            return self.async_abort(reason="code_expired")
+        return await super().async_step_creation(user_input)
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create an entry for the flow, or update existing entry."""
+        existing_entries = self._async_current_entries()
+        if existing_entries:
+            updated = False
+            for entry in existing_entries:
+                if updated:
+                    await self.hass.config_entries.async_remove(entry.entry_id)
```

Comment:
Why would there be other config entries to remove?

Reply:
This is not expected. I mis-remembered our conversation in https://github.com/home-assistant/core/pull/44202#discussion_r546022110 and why nest had this.  Removed.